theano/gof/cc.py

```diff
@@ -287,21 +287,23 @@ def struct_gen(args, struct_builders, blocks, sub):
             %(struct_init_head)s
             this->__ERROR = __ERROR;
             return 0;
-        }
+        }// end init()
         void cleanup(void) {
             %(struct_cleanup)s
             %(storage_decref)s
-        }
-        int run(void) {
+        } // end cleanup()
+        int run(void);
+    };
+    """ % sub
+    run_code = """
+        DllExport int %(name)s::run(void) {
```

Comment:
Yes, I think we can export the run method only in the case the function is compiled as shared library